lib/services/storage.dart

```diff
@@ -88,156 +83,175 @@ class Storage extends Service {
+    /// Get File Preview
+    ///
+    /// Get a file preview image. Currently, this method supports preview for image
+    /// files (jpg, png, and gif), other supported formats, like pdf, docs, slides,
+    /// and spreadsheets, will return the file icon image. You can also pass query
+    /// string arguments for cutting and resizing your preview image. Preview is
+    /// supported only for image files smaller than 10MB.
+    ///
+    Uri getFilePreview({required String bucketId, required String fileId, int? width, int? height, String? gravity, int? quality, int? borderWidth, String? borderColor, int? borderRadius, double? opacity, int? rotation, String? background, String? output}) {
         final String path = '/storage/buckets/{bucketId}/files/{fileId}/preview'.replaceAll('{bucketId}', bucketId).replaceAll('{fileId}', fileId);
 
         final Map<String, dynamic> params = {
             'width': width,
-'height': height,
-'gravity': gravity,
-'quality': quality,
-'borderWidth': borderWidth,
-'borderColor': borderColor,
-'borderRadius': borderRadius,
-'opacity': opacity,
-'rotation': rotation,
-'background': background,
-'output': output,
-
-            
+            'height': height,
+            'gravity': gravity,
+            'quality': quality,
+            'borderWidth': borderWidth,
+            'borderColor': borderColor,
+            'borderRadius': borderRadius,
+            'opacity': opacity,
+            'rotation': rotation,
+            'background': background,
+            'output': output,
             'project': client.config['project'],
         };
+        final uri = Uri.parse(client.endpoint + path).replace(queryParameters: params);
 
-        final res = await client.call(HttpMethod.get, path: path, params: params, responseType: ResponseType.bytes);
-        return res.data;
+        return uri;
+    }
 
+    Future<Uint8List> getFilePreviewBytes({required String bucketId, required String fileId, int? width, int? height, String? gravity, int? quality, int? borderWidth, String? borderColor, int? borderRadius, double? opacity, int? rotation, String? background, String? output}) async {
+        final uri = getFilePreview(
+            bucketId: bucketId,
+            fileId: fileId,
+            width: width,
+            height: height,
+            gravity: gravity,
+            quality: quality,
+            borderWidth: borderWidth,
+            borderColor: borderColor,
+            borderRadius: borderRadius,
+            opacity: opacity,
+            rotation: rotation,
+            background: background,
+            output: output,
+        );
+
+        final res = await client.call(HttpMethod.get, path: uri.path[1:], params: uri.queryParameters, responseType: ResponseType.bytes);
```

Comment:
Oops! `uri.path[1:]` is python syntax. We could do something like:

```suggestion
        final path = "/" + uri.pathSegments.sublist(1).join("/");

        final res = await client.call(HttpMethod.get, path: path, params: uri.queryParameters, responseType: ResponseType.bytes);
```